### DISCUSSION TITLE
Proxy function(s): Ease

### DISCUSSION THREAD

**Discussion by iskawhiskers** [CapsAdmin/pac3#1299] [Ideas]
A maths library for easing exists in GLua already (https://wiki.facepunch.com/gmod/math.ease) and is used in the custom animation part, but is not used in the proxy part.  Would be useful to have proxy functions that use this library in a similar manner to lerp, in cases where a value not already handled by custom animation can be animated in a specific way over a period of time.

Either the functions could be individually defined based on their type:
easeInSine(fraction, a, b)
easeOutSine(fraction, a, b)
easeInOutSine(fraction, a, b)
etc.

or it could be a combined function that adds an extra variable:
ease(easetype, fraction, a, b)

**Comment by pingu7867** [comment]
added today
